File: Java/src/main/java/br/com/locaweb/locamail/api/model/Usuario.java
package br.com.locaweb.locamail.api.model;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "T_LCW_USUARIO",
        indexes = @Index(columnList = "email", unique = true))
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@EqualsAndHashCode
public class Usuario {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id_usuario")
    private Long id_usuario;

    @Column(name = "email", nullable = false, unique = true)
    private String email;

    @Column(name = "nome", nullable = false)
    private String nome;

    @Column(name = "senha", nullable = false)
    private String senha;

    @Column(name = "autenticado", nullable = false)
    private Boolean autenticado;

    @Column(name = "profile_image", nullable = false)
    @Lob
    private byte[] profile_image;

    @Column(name = "selected_user", nullable = false)
    private Boolean selected_user;
}

File: Kotlin/app/src/main/java/br/com/fiap/locawebmailapp/model/Usuario.kt
package br.com.fiap.locawebmailapp.model

import androidx.compose.runtime.Immutable
import androidx.compose.runtime.Stable
import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.Index
import androidx.room.PrimaryKey
import java.time.LocalDate
import java.time.LocalDateTime
import java.time.LocalTime
import java.time.format.DateTimeFormatter

@Immutable
@Entity(tableName = "T_LCW_USUARIO",
    indices = [Index(value = ["email"], unique = true)])
data class Usuario(
    @PrimaryKey(autoGenerate = true) var id_usuario: Long = 0,
    @ColumnInfo(name = "email") var email: String,
    var nome: String,
    var senha: String,
    var autenticado: Boolean,
    var profile_image: ByteArray = ByteArray(10),
    var selected_user: Boolean
)

@Immutable
data class UsuarioSemSenha(
    var id_usuario: Long = 0,
    var email: String,
    var nome: String,
    var autenticado: Boolean,
    var profile_image: ByteArray = ByteArray(10),
    var selected_user: Boolean
)